src/bridges/dca/BiDCABridge.sol

```diff
@@ -0,0 +1,648 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IRollupProcessor} from "../../aztec/interfaces/IRollupProcessor.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+
+import {IWETH} from "../../interfaces/IWETH.sol";
+
+import {SafeCastLib} from "./SafeCastLib.sol";
+
+/**
+ * @notice Initial abstract implementation of "Dollar" Cost Averaging.
+ * Supports bidirectional "selling", e.g., A -> B and B -> A in same bridge
+ * Will match internal orders first, before allowing external parties to trade.
+ * @dev Built for assets with 18 decimals precision
+ * @dev A contract that inherits must handle the case for forcing a swap through a DEX.
+ * @author Lasse Herskind (LHerskind on GitHub).
+ */
```

Comment:
Can you please describe here how the bridge internally works? How the ticks are indexed by time, how does rebalancing work etc.